src/tomocupy/retrieve_phase.py

```diff
@@ -195,6 +211,28 @@ def _reciprocal_grid(pixel_size, nx, ny):
     # there is no substitute for np.add.outer using cupy.
     return cp.array(np.add.outer(indx, indy))
 
+def _reciprocal_gridG(pixel_size, nx, ny):
+    """
+    Calculate reciprocal grid for Generalized Paganin method.
+
+    Parameters
+    ----------
+    pixel_size : float
+        Detector pixel size in cm.
+    nx, ny : int
+        Size of the reciprocal grid along x and y axes.
+
+    Returns
+    -------
+    ndarray
+        Grid coordinates.
+    """
+    # Considering diffracting feature ~2*pixel size
+    # Sampling in reciprocal space.
+    indx = np.cos(_reciprocal_coord(pixel_size, nx)*2*PI*pixel_size)
+    indy = np.cos(_reciprocal_coord(pixel_size, ny)*2*PI*pixel_size)
+    
+    return np.add.outer(indx, indy)
```

Comment:
I believe you can do it like this if indx and indy are one-dimensional:
[indxx,indyy]=np.meshgrid(indy,indx)
indxx+indyy

Reply:
Modified both retrieval methods fully in cupy